File: apps/web-shop/src/types/order.ts
export type OrderStatus = "Pending" | "Processing" | "Shipped" | "Delivered" | "Cancelled";
export type PaymentMethod = "CashOnDelivery" | "CreditCard" | "MockPayment";

export interface OrderItemDto {
  id: string;
  productId: string;
  productName: string;
  productSKU: string;
  unitPrice: number;
  quantity: number;
  totalPrice: number;
}

export interface OrderDto {
  id: string;
  orderNumber: string;
  status: OrderStatus;
  shippingRecipientName: string;
  shippingStreet: string;
  shippingCity: string;
  shippingProvince: string;
  shippingPostalCode: string;
  shippingPhone: string;
  subtotal: number;
  shippingFee: number;
  tax: number;
  totalAmount: number;
  paymentMethod: string;
  paymentStatus: string;
  createdAt: string;
  items: OrderItemDto[];
}

export interface CheckoutSummaryDto {
  subtotal: number;
  shippingFee: number;
  tax: number;
  totalAmount: number;
  totalItems: number;
}

export interface ShippingAddressRequest {
  recipientName: string;
  street: string;
  city: string;
  province: string;
  postalCode: string;
  phone: string;
}

export interface CreateOrderRequest {
  shippingAddress: ShippingAddressRequest;
  paymentMethod: PaymentMethod;
  /**
   * Ids of the cart items to check out. When omitted or empty, the entire cart
   * is ordered (backward compatible). When provided, only these items are
   * ordered and removed from the cart; the rest stay in the cart.
   */
  selectedItemIds?: string[];
}

File: apps/api-oos/Models/Order.cs
namespace ApiOos.Models;

using ApiOos.Enums;


public class Order
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string OrderNumber { get; set; } = string.Empty;
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    // Shipping address info (snapshot)
    public string ShippingRecipientName { get; set; } = string.Empty;
    public string ShippingStreet { get; set; } = string.Empty;
    public string ShippingCity { get; set; } = string.Empty;
    public string ShippingProvince { get; set; } = string.Empty;
    public string ShippingPostalCode { get; set; } = string.Empty;
    public string ShippingPhone { get; set; } = string.Empty;

    // Pricing breakdown
    public decimal Subtotal { get; set; }
    public decimal ShippingFee { get; set; }
    public decimal Tax { get; set; }
    public decimal TotalAmount { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
    public Payment? Payment { get; set; }
}
